include/deal.II/lac/petsc_matrix_base.h

```diff
@@ -784,6 +784,44 @@ namespace PETScWrappers
     void Tvmult_add (VectorBase       &dst,
                      const VectorBase &src) const;
 
+    /**
+     * Perform the matrix-matrix multiplication $C = AB$, or, if an
+     * vector $V$ whose size is compatible with B is given,
+     * $C = A \text{diag}(V) B$, where $\text{diag}(V)$ defines a
+     * diagonal matrix with the vector entries.
+     *
+     * This function assumes that the calling matrix $A$ and $B$
+     * have compatible sizes. The size of $C$ will be set within this
+     * function.
+     *
+     * The content as well as the sparsity pattern of the matrix $C$ will be
```

Comment:
talking about the sparsity pattern of the matrix implies that you are specifically thinking of sparse matrices. But then this function should be in the sparse matrix class.